torch/package/package_importer.py

```diff
@@ -82,14 +82,17 @@ def __init__(
         self.root = _PackageNode(None)
         self.modules = {}
         self.extern_modules = self._read_extern()
-
+        self.selective_extern_packages = self._read_selective_extern()
         for extern_module in self.extern_modules:
             if not module_allowed(extern_module):
                 raise ImportError(
                     f"package '{file_or_buffer}' needs the external module '{extern_module}' "
                     f"but that module has been disallowed"
                 )
-            self._add_extern(extern_module)
+            if extern_module in self.selective_extern_packages:
```

Comment:
I find this logic kind of confusing. Why are there `selective_extern` modules to the `extern` list? Shouldn't these lists be unrelated to each other?

Reply:
Yeah this was for an older pattern which I was worried about where the user would call _selective_intern on the same package twice. If we are not using that functionality, I don't think selective extern packages should randomly end up in the extern list anymore.